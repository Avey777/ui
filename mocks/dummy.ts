export const source = {
  id: '2',
  name: 'minikube-influx',
  type: 'influx',
  url: 'http://192.168.99.100:30400',
  'default': true,
  telegraf: 'telegraf',
  organization: 'default',
  role: 'viewer',
  links: {
    self: '/chronograf/v1/sources/2',
    kapacitors: '/chronograf/v1/sources/2/kapacitors',
    proxy: '/chronograf/v1/sources/2/proxy',
    queries: '/chronograf/v1/sources/2/queries',
    write: '/chronograf/v1/sources/2/write',
    permissions: '/chronograf/v1/sources/2/permissions',
    users: '/chronograf/v1/sources/2/users',
    databases: '/chronograf/v1/sources/2/dbs',
    annotations: '/chronograf/v1/sources/2/annotations'
  }
}

export const kapacitor = {
  id: '1',
  name: 'Test Kapacitor',
  url: 'http://localhost:9092',
  insecureSkipVerify: false,
  active: true,
  links: {
    self: '/chronograf/v1/sources/47/kapacitors/1',
    proxy: '/chronograf/v1/sources/47/kapacitors/1/proxy',
  },
}

export const createKapacitorBody = {
  name: 'Test Kapacitor',
  url: 'http://localhost:9092',
  insecureSkipVerify: false,
  username: 'user',
  password: 'pass',
}
