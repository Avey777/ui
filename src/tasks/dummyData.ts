import {Task, TaskStatus} from 'src/types/v2/tasks'

export const dummyTasks: Task[] = [
  {
    id: '02ef9deff2141000',
    organizationId: '02ee9e2a29d73000',
    name: 'pasdlak',
    status: TaskStatus.Active,
    owner: {id: '02ee9e2a19d73000', name: ''},
    flux:
      'option task = {\n  name: "pasdlak",\n  cron: "2 0 * * *"\n}\nfrom(bucket: "inbucket") \n|> range(start: -1h)',
    cron: '2 0 * * *',
    organization: {
      links: {
        buckets: '/api/v2/buckets?org=RadicalOrganization',
        dashboards: '/api/v2/dashboards?org=RadicalOrganization',
        log: '/api/v2/orgs/02ee9e2a29d73000/log',
        members: '/api/v2/orgs/02ee9e2a29d73000/members',
        self: '/api/v2/orgs/02ee9e2a29d73000',
        tasks: '/api/v2/tasks?org=RadicalOrganization',
      },
      id: '02ee9e2a29d73000',
      name: 'RadicalOrganization',
    },
  },
  {
    id: '02f12c50dba72000',
    organizationId: '02ee9e2a29d73000',
    name: 'somename',
    status: TaskStatus.Active,
    owner: {id: '02ee9e2a19d73000', name: ''},
    flux:
      'option task = {\n  name: "somename",\n  every: 1m,\n}\nfrom(bucket: "inbucket") \n|> range(start: -task.every)',
    every: '1m0s',
    organization: {
      links: {
        buckets: '/api/v2/buckets?org=RadicalOrganization',
        dashboards: '/api/v2/dashboards?org=RadicalOrganization',
        log: '/api/v2/orgs/02ee9e2a29d73000/log',
        members: '/api/v2/orgs/02ee9e2a29d73000/members',
        self: '/api/v2/orgs/02ee9e2a29d73000',
        tasks: '/api/v2/tasks?org=RadicalOrganization',
      },
      id: '02ee9e2a29d73000',
      name: 'RadicalOrganization',
    },
  },
]
