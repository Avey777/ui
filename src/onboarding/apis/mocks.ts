import {telegrafConfigsResponse, authResponse} from 'mocks/dummyData'

const telegrafsGet = jest.fn(() => Promise.resolve(telegrafConfigsResponse))
const authorizationsGet = jest.fn(() => Promise.resolve(authResponse))

export const telegrafsAPI = {
  telegrafsGet,
}

export const authorizationsAPI = {
  authorizationsGet,
}
