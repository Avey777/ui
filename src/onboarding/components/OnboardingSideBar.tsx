import React, {Component} from 'react'

// Components
import SideBar from 'src/onboarding/components/side_bar/SideBar'
import {SideBarTabStatus as TabStatus} from 'src/onboarding/components/side_bar/SideBar'
import {ComponentColor} from 'src/clockface'

// Utils
import {downloadTextFile} from 'src/shared/utils/download'

// Constants
import {getTelegrafConfigFailed} from 'src/shared/copy/v2/notifications'

// APIs
import {getTelegrafConfigTOML, createTelegrafConfig} from 'src/onboarding/apis'

// Types
import {IconFont} from 'src/clockface'
import {DataSource, ConfigurationState} from 'src/types/v2/dataLoaders'
import {NotificationAction} from 'src/types'

interface Props {
  title: string
  visible: boolean
  dataSources: DataSource[]
  notify: NotificationAction
  onTabClick: (tabID: string) => void
}

const configStateToTabStatus = (cs: ConfigurationState): TabStatus => {
  switch (cs) {
    case ConfigurationState.Unconfigured:
      return TabStatus.Default
    case ConfigurationState.Verifying:
      return TabStatus.Pending
    case ConfigurationState.Configured:
      return TabStatus.Default
    case ConfigurationState.Loading:
      return TabStatus.Pending
    case ConfigurationState.Done:
      return TabStatus.Success
    case ConfigurationState.Error:
      return TabStatus.Error
  }
}

class OnboardingSideBar extends Component<Props> {
  public render() {
    const {title, visible} = this.props
    return (
      <SideBar title={title} visible={visible}>
        {[...this.tabs, ...this.buttons]}
      </SideBar>
    )
  }

  private get tabs(): JSX.Element[] {
    const {dataSources, onTabClick} = this.props
    return dataSources.map(t => (
      <SideBar.Tab
        label={t.name}
        key={t.name}
        id={t.name}
        active={t.active}
        status={configStateToTabStatus(t.configured)}
        onClick={onTabClick}
      />
    ))
  }

  private get buttons(): JSX.Element[] {
    return [
      <SideBar.Button
        key="Download Config File"
        text="Download Config File"
        titleText="Download Config File"
        color={ComponentColor.Secondary}
        icon={IconFont.Download}
        onClick={this.handleDownload}
      />,
      <SideBar.Button
        text="Add New Source"
        key="Add New Source"
        titleText="Add New Source"
        color={ComponentColor.Default}
        icon={IconFont.Plus}
      />,
    ]
  }

  private handleDownload = async () => {
    const {notify} = this.props
    try {
      const telegraf = await createTelegrafConfig()
      const config = await getTelegrafConfigTOML(telegraf.id)
      downloadTextFile(config, 'config.toml')
    } catch (error) {
      notify(getTelegrafConfigFailed())
    }
  }
}

export default OnboardingSideBar
