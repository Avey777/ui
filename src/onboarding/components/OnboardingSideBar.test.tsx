// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import OnboardingSideBar from 'src/onboarding/components/OnboardingSideBar'

import {dataSources} from 'mocks/dummyData'

const onClick = jest.fn(() => {})

const setup = (override?) => {
  const props = {
    title: 'title',
    visible: true,
    dataSources,
    onTabClick: onClick,
    ...override,
  }

  const wrapper = shallow(<OnboardingSideBar {...props} />)

  return {wrapper}
}

describe('OnboardingSideBar', () => {
  describe('rendering', () => {
    it('renders! wee!', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
