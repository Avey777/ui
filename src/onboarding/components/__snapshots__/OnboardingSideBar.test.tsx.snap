// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingSideBar rendering renders! wee! 1`] = `
<SideBar
  title="title"
  visible={true}
>
  <SideBarTab
    active={true}
    id="bestDataSource"
    key="bestDataSource"
    label="bestDataSource"
    onClick={[MockFunction]}
    status="default"
  />
  <SideBarTab
    active={true}
    id="secondBestDataSource"
    key="secondBestDataSource"
    label="secondBestDataSource"
    onClick={[MockFunction]}
    status="success"
  />
  <SideBarButton
    color="secondary"
    icon="download"
    key="Download Config File"
    onClick={[Function]}
    text="Download Config File"
    titleText="Download Config File"
  />
  <SideBarButton
    color="default"
    icon="plus"
    key="Add New Source"
    text="Add New Source"
    titleText="Add New Source"
  />
</SideBar>
`;
